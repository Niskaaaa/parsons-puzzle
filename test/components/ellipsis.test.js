import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtEllipsis from '../../.temp/components/ellipsis/index'

describe('AtDivider Snap', () => {
  it('render AtEllipsis -- props lines', () => {
    const component = renderToString(
      <AtEllipsis lines={2}>测试测试测试测试</AtEllipsis>
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtEllipsis -- props text', () => {
    const component = renderToString(
      <AtEllipsis text='测试测试测试测试测试' lines={2} />
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtDivider -- props expand', () => {
    const component = renderToString(
      <AtEllipsis text='测试测试测试测试测试' lines={2} expand />
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtDivider -- props btnText', () => {
    const component = renderToString(
      <AtEllipsis
        text='测试测试测试测试测试'
        lines={2}
        expand
        btnText='查看全部'
      />
    )
    expect(component).toMatchSnapshot()
  })
})
