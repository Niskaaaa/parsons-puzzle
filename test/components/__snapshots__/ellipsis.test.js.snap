// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtDivider Snap render AtDivider -- props btnText 1`] = `"<div class=\\"\\"><div class=\\"taro-canvas\\" style=\\"position:fixed;left:-1000px;top:-1000px;width:100%;height:100%;\\"><canvas canvasId=\\"canvas\\" width=\\"300\\" height=\\"150\\"></canvas></div><div id=\\"indexes-ellipsis-AOTU2018\\" class=\\"at-ellipsis at-ellipsis--WEB at-ellipsis--lineclamp\\"><div style=\\"font-size:14px;color:#646A73;\\" class=\\"at-ellipsis__block\\">测试测试测试测试测试</div>undefinedundefined</div></div>"`;

exports[`AtDivider Snap render AtDivider -- props expand 1`] = `"<div class=\\"\\"><div class=\\"taro-canvas\\" style=\\"position:fixed;left:-1000px;top:-1000px;width:100%;height:100%;\\"><canvas canvasId=\\"canvas\\" width=\\"300\\" height=\\"150\\"></canvas></div><div id=\\"indexes-ellipsis-AOTU2018\\" class=\\"at-ellipsis at-ellipsis--WEB at-ellipsis--lineclamp\\"><div style=\\"font-size:14px;color:#646A73;\\" class=\\"at-ellipsis__block\\">测试测试测试测试测试</div>undefinedundefined</div></div>"`;

exports[`AtDivider Snap render AtEllipsis -- props lines 1`] = `"<div id=\\"indexes-ellipsis-AOTU2018\\" style=\\"-webkit-line-clamp:2;\\" class=\\"at-ellipsis at-ellipsis--WEB at-ellipsis--lineclamp\\">测试测试测试测试</div>"`;

exports[`AtDivider Snap render AtEllipsis -- props text 1`] = `"<div class=\\"\\"><div class=\\"taro-canvas\\" style=\\"position:fixed;left:-1000px;top:-1000px;width:100%;height:100%;\\"><canvas canvasId=\\"canvas\\" width=\\"300\\" height=\\"150\\"></canvas></div><div id=\\"indexes-ellipsis-AOTU2018\\" class=\\"at-ellipsis at-ellipsis--WEB at-ellipsis--lineclamp\\"><div style=\\"font-size:14px;color:#646A73;\\" class=\\"at-ellipsis__block\\">测试测试测试测试测试</div>undefined<div style=\\"font-size:14px;color:#646A73;\\" class=\\"\\">测试测试测试测试测试</div>undefined</div></div>"`;
